storage/flux/reader.go

```diff
@@ -218,6 +246,276 @@ READ:
 	return rs.Err()
 }
 
+type windowAggregateIterator struct {
+	ctx   context.Context
+	s     storage.Store
+	spec  query.ReadWindowAggregateSpec
+	stats cursors.CursorStats
+	cache *tagsCache
+	alloc *memory.Allocator
+}
+
+func (wai *windowAggregateIterator) Statistics() cursors.CursorStats { return wai.stats }
+
+func (wai *windowAggregateIterator) Do(f func(flux.Table) error) error {
+	src := wai.s.GetSource(
+		uint64(wai.spec.OrganizationID),
+		uint64(wai.spec.BucketID),
+	)
+
+	// Setup read request
+	any, err := types.MarshalAny(src)
+	if err != nil {
+		return err
+	}
+
+	var req datatypes.ReadWindowAggregateRequest
+	req.ReadSource = any
+	req.Predicate = wai.spec.Predicate
+	req.Range.Start = int64(wai.spec.Bounds.Start)
+	req.Range.End = int64(wai.spec.Bounds.Stop)
+
+	if wai.spec.WindowEvery != 0 || wai.spec.Offset != 0 {
+		req.Window = &datatypes.Window{
+			Every: &datatypes.Duration{
+				Nsecs: wai.spec.WindowEvery,
+			},
+			Offset: &datatypes.Duration{
+				Nsecs: wai.spec.Offset,
+			},
+		}
+	} else {
+		req.Window = &datatypes.Window{
+			Every: &datatypes.Duration{
+				Nsecs:    wai.spec.Window.Every.Nanoseconds(),
+				Months:   wai.spec.Window.Every.Months(),
+				Negative: wai.spec.Window.Every.IsNegative(),
+			},
+			Offset: &datatypes.Duration{
+				Nsecs:    wai.spec.Window.Offset.Nanoseconds(),
+				Months:   wai.spec.Window.Offset.Months(),
+				Negative: wai.spec.Window.Offset.IsNegative(),
+			},
+		}
+	}
+
+	req.Aggregate = make([]*datatypes.Aggregate, len(wai.spec.Aggregates))
```

Comment:
`datatypes` was substituted for anywhere that `storageproto` was previously used.